tests/semantic_analyser.cpp

```diff
@@ -500,6 +500,8 @@ TEST(semantic_analyser, ternary_expressions)
   test(R"(kprobe:f { @x = pid < 10000 ? "lo" : "high" })");
   test("kprobe:f { pid < 10000 ? printf(\"lo\") : exit() }");
   test("kprobe:f { curtask ? printf(\"lo\") : exit() }");
+  test(R"(kprobe:f { $x = pid < 10000 ? buf("mystr", 5) : buf("mystr", 4) })");
+  test(R"(kprobe:f { $x = pid < 10000 ? macaddr(arg0) : macaddr(arg1); })");
```

Comment:
Maybe table driven test with all the types?

Reply:
Feel like a bit of overkill since we don't have separate branches for everything that's not a string, int, or none but I'll look into it.

Reply:
Yeah, perhaps overkill. But upshot is it encodes intent to support all types

Reply:
Added a lot more types and support for different sized tuples.